web/src/routes/Event.jsx

```diff
@@ -8,6 +13,23 @@ import { Table, Thead, Tbody, Th, Tr, Td } from '../components/Table';
 export default function Event({ eventId }) {
   const apiHost = useApiHost();
   const { data, status } = useEvent(eventId);
+  const [showDialog, setShowDialog] = useState(false);
+
+  const handleClickDelete = () => {
+    setShowDialog(true);
+  }
+
+  const handleDismissDeleteDialog = () => {
+    setShowDialog(false);
+  };
+
+  const handleClickDeleteDialog = useCallback(
+    async () => {
+      await fetch(`${apiHost}/api/events/${eventId}`, {method: 'DELETE'});
```

Comment:
if the eventId changes before the fetch has finished, we'll get some weird behavior here. you use useCallback here but no where else. will defeat the purpose since there will still be rerenders